gitlab/config.py

```diff
@@ -66,155 +93,149 @@ def __init__(
         self, gitlab_id: Optional[str] = None, config_files: Optional[List[str]] = None
     ) -> None:
         self.gitlab_id = gitlab_id
-        _files = config_files or _DEFAULT_FILES
-        file_exist = False
-        for file in _files:
-            if os.path.exists(file):
-                file_exist = True
-        if not file_exist:
-            raise GitlabConfigMissingError(
-                "Config file not found. \nPlease create one in "
-                "one of the following locations: {} \nor "
-                "specify a config file using the '-c' parameter.".format(
-                    ", ".join(_DEFAULT_FILES)
-                )
-            )
+        self.http_username: Optional[str] = None
+        self.http_password: Optional[str] = None
+        self.job_token: Optional[str] = None
+        self.oauth_token: Optional[str] = None
+        self.private_token: Optional[str] = None
+
+        self.api_version: str = "4"
+        self.order_by: Optional[str] = None
+        self.pagination: Optional[str] = None
+        self.per_page: Optional[int] = None
+        self.retry_transient_errors: bool = False
+        self.ssl_verify: Union[bool, str] = True
+        self.timeout: int = 60
+        self.url: str = DEFAULT_URL
+        self.user_agent: str = USER_AGENT
 
-        self._config = configparser.ConfigParser()
-        self._config.read(_files)
+        self._files = _get_config_files(config_files)
+        if self._files:
+            self._parse_config()
+
+    def _parse_config(self) -> None:
+        _config = configparser.ConfigParser()
+        _config.read(self._files)
 
         if self.gitlab_id is None:
             try:
-                self.gitlab_id = self._config.get("global", "default")
+                self.gitlab_id = _config.get("global", "default")
             except Exception as e:
                 raise GitlabIDError(
                     "Impossible to get the gitlab id (not specified in config file)"
                 ) from e
 
         try:
-            self.url = self._config.get(self.gitlab_id, "url")
+            self.url = _config.get(self.gitlab_id, "url")
```

Comment:
Does this work if there is no config file?

Reply:
Thanks for that! Added some logic that more explicitly fails on CLI args. There's a functional test for that as well now.